### PR TITLE
Add Okta and Azure AD set up docs

### PR DESCRIPTION
**Pull request by chomipi88**

### CODE REVIEW

File: cloud/configure-idp.md [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+title: "Set Up an Identity Provider"
+id: configure-idp
+description: Configure federated authentication from a variety of third party identity providers on Astronomer Cloud.
+---
+
+## Overview
+
+This guide provides setup steps for integrating various third party identity providers (IDP) with Astronomer Cloud.
+
+After integrating your identity provider with Astronomer Cloud, a user in your organization will be automatically logged in to Astronomer Cloud if they're already logged in via your IDP. You can also add existing IDP accounts as new Astronomer users through your IDP application.
```
Comment (jwitz) [inline cloud/configure-idp.md:12]: @chomipi88 Is this correct? If so, do we need to edit our "Add User" documentation to mention this?

File: cloud/configure-idp.md [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+
+This section provides setup steps for setting up Okta as your IDP on Astronomer Cloud. After completing this setup, your organization's users can use Okta to log in to Astronomer.
+
+### Prerequisites
+
+To integrate Okta as your IDP for Astronomer Cloud, you must have an [Okta account](https://www.okta.com/)
+
+### Step 1: Work with Astronomer to configure SAML
+
+Reach out to [Astronomer Support](support.astronomer.io) with a request to integrate Okta as an IDP on Astronomer. From here, an Astronomer Representative will provide you with a connection name. Save this connection name for use in Step 2.
```
Comment (jwitz) [inline cloud/configure-idp.md:24]: @chomipi88 Is this the only info that Astronomer provides at this stage? Assumed it from the configurations below

File: cloud/configure-idp.md (hunk 1 above)
Comment (chomipi88) [reply cloud/configure-idp.md:12]: @jwitz I don't quite understand what this is saying.
What it expected, is post configuration of the IdP, the customers' users will be able to log into Astronomer Cloud (via Cloud UI) as long as they have a valid user account on their IdP.

File: cloud/configure-idp.md (hunk 2 above)
Comment (chomipi88) [reply cloud/configure-idp.md:24]: @jwitz technically we will provide the full Single Sign On URL and Audience URI which will include their connection name

**Comment by chomipi88** [comment]
Looks good otherwise

File: cloud/configure-idp.md (hunk 1 above)
Comment (jwitz) [reply cloud/configure-idp.md:12]: So what you're saying is that by using an IDP, you no longer have to do [Step 1](https://docs.astronomer.io/cloud/add-user#step-1-invite-the-user-to-astronomer) here. The email is automatically associated with the organization, so a Workspace admin could add a user to a Workspace via email without them needing to properly register with Astronomer.

Comment (jwitz) [reply cloud/configure-idp.md:12]: @chomipi88 I assume there's still some control in the IDP of which users belong to which apps. So is there a prerequisite of needing to add users to the Astronomer app via the IDP before they can authenticate to Astronomer

Comment (chomipi88) [reply cloud/configure-idp.md:12]: @jwitz Definitely a different experience then the linked Step 1. The user will be logged into their Organization. From there they will be able to create a workspace (shipping soon) or be added to an existing one.

File: cloud/configure-idp.md [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+title: "Set Up an Identity Provider"
+id: configure-idp
+description: Configure federated authentication from a variety of third party identity providers on Astronomer Cloud.
+---
+
+## Overview
+
+This guide provides setup steps for integrating various third party identity providers (IDP) with Astronomer Cloud.
+
+After integrating your identity provider with Astronomer Cloud, a user in your organization will be automatically logged in to Astronomer Cloud if they're already logged in via your IDP. By adding new Astronomer users through your IDP's own user management system, Workspace admins can automatically add new users to their Workspace without those users needing to individually sign up for Astronomer.
```
Comment (paolaperaza) [inline cloud/configure-idp.md:12]:
```suggestion
Identity Providers (IdPs) are services that manage user accounts. As organizations grow, it's common for teams to integrate internal tooling with a third-party IdP. This allows administrators to monitor application access, user permissions, and security policies from a single place. It also makes it easy for individual users to access the tools they need.

Astronomer Cloud supports integrations with the following IdPs:

- [Okta](https://www.okta.com/)
- [Azure Active Directory (AD)](https://azure.microsoft.com/en-us/services/active-directory/)

This guide provides setup steps for integrating both of these identity providers on Astronomer.

Once you complete the integration for your organization,
- Astronomer will be listed in your IdP's set of supported applications.
- Your IdP will be listed as an authentication method in the log in page of the Astronomer UI.
- Workspace Admins can invite a user to a Workspace directly without that user having to separately sign up for an account first.
- Users will automatically be authenticated to Astronomer if they're already logged in to your IDP.
```

@jwitz FYI I think the right acronym is `IdP` not `IDP`. That's what Okta uses: https://help.okta.com/en/prod/Content/Topics/Security/Identity_Providers.htm

File: cloud/configure-idp.md [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+
+## Configure Okta as Your IDP
```
Comment (paolaperaza) [inline cloud/configure-idp.md:14]:
```suggestion
## Configure Okta as Your Identity Provider
```

File: cloud/configure-idp.md [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+
+This section provides setup steps for setting up Okta as your IDP on Astronomer Cloud. After completing this setup, your organization's users can use Okta to log in to Astronomer.
```
Comment (paolaperaza) [inline cloud/configure-idp.md:16]:
```suggestion
This section provides setup steps for setting up Okta as your IDP on Astronomer Cloud. After completing this setup, all users in your organization can use Okta to log in to Astronomer.
```

Comment (paolaperaza) [reply cloud/configure-idp.md:16]: @chomipi88 Is this dependent on users having the same email base domain that's set for the organization? Should we mention something about that here?

File: cloud/configure-idp.md [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+
+### Prerequisites
+
+To integrate Okta as your IDP for Astronomer Cloud, you must have an [Okta account](https://www.okta.com/)
```
Comment (paolaperaza) [inline cloud/configure-idp.md:20]:
```suggestion
To integrate Okta as your IDP for Astronomer Cloud, you must have an [Okta account](https://www.okta.com/).
```

Any role required?

File: cloud/configure-idp.md [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+
+### Step 1: Work with Astronomer to configure SAML
+
+Reach out to [Astronomer Support](support.astronomer.io) with a request to integrate Okta as an IDP on Astronomer. From here, an Astronomer Representative will provide you with an SSO URL and an Audience URI. Save these values for Step 2.
```
Comment (paolaperaza) [inline cloud/configure-idp.md:24]:
```suggestion
Reach out to [Astronomer Support](https://support.astronomer.io) with a request to integrate Okta as an IdP on Astronomer. From here, an Astronomer Representative will provide you with two things:

- A Single Sign-On (SSO) URL
- An Audience URI

Save these values for Step 2.
```

File: cloud/configure-idp.md [hunk 8]
```diff
@@ -0,0 +1,120 @@
...
+
+## Configure Okta as Your IDP
+
+This section provides setup steps for setting up Okta as your IDP on Astronomer Cloud. After completing this setup, your organization's users can use Okta to log in to Astronomer.
+
+### Prerequisites
+
+To integrate Okta as your IDP for Astronomer Cloud, you must have an [Okta account](https://www.okta.com/)
+
+### Step 1: Work with Astronomer to configure SAML
```
Comment (paolaperaza) [inline cloud/configure-idp.md:22]:
```suggestion
### Step 1: Reach out to Astronomer
```

File: cloud/configure-idp.md [hunk 9]
```diff
@@ -0,0 +1,120 @@
...
+
+Reach out to [Astronomer Support](support.astronomer.io) with a request to integrate Okta as an IDP on Astronomer. From here, an Astronomer Representative will provide you with an SSO URL and an Audience URI. Save these values for Step 2.
+
+### Step 2: Configure Okta
+
+1. Follow the [Okta documentation](https://help.okta.com/en/prod/Content/Topics/Apps/Apps_App_Integration_Wizard_SAML.htm) to create an SAML App Integration with the following SAML settings:
```
Comment (paolaperaza) [inline cloud/configure-idp.md:28]:
```suggestion
1. Follow the [Okta documentation](https://help.okta.com/en/prod/Content/Topics/Apps/Apps_App_Integration_Wizard_SAML.htm) to create a SAML App Integration via Okta's Admin Console. Configure the following SAML settings:
```

File: cloud/configure-idp.md [hunk 10]
```diff
@@ -0,0 +1,120 @@
...
+
+    - **Single sign on URL**: `<your-sso-url>`
+    - **Audience URI (SP Entity ID)**: `<your-audience-uri>`
+    - **Name ID format**: `Unspecified`
+    - **Application username**: `Email`
+    - **Update application username on**: `Create and update`
```
Comment (paolaperaza) [inline cloud/configure-idp.md:34]:
```suggestion
    - **Single Sign-On URL**: `<your-sso-url>`
    - **Audience URI (SP Entity ID)**: `<your-audience-uri>`
    - **Name ID format**: `Unspecified`
    - **Application username**: `Email`
    - **Update application username on**: `Create and update`
```

Comment (paolaperaza) [reply cloud/configure-idp.md:34]: Any guidance on what this email can be?

File: cloud/configure-idp.md [hunk 11]
```diff
@@ -0,0 +1,120 @@
...
+
+  These values might be different if Okta is connected to an Active Directory. In this case, replace each `Value` with the equivalent Active Directory values for a user's first name, last name, and full email address.
+
+  :::
+
+4. Complete the remainder of the setup as documented in Okta until you finish creating your integration.
+
+### Step 3: Provide Astronomer with your integration information
+
+On the page for your Okta app integration, click **View Setup Instructions**. Copy the values for `Single Sign-on URL` and `X.509 Certificate` that appear and send them to Astronomer Support. From here, Astronomer will finalize the integration with Okta.
```
Comment (paolaperaza) [inline cloud/configure-idp.md:63]:
```suggestion
On the page for your Okta app integration, click **View Setup Instructions**. Copy the values for `Single Sign-on URL` and `X.509 Certificate` that appear and send them to Astronomer Support.

From here, Astronomer will finalize your organization's integration with Okta.
```

File: cloud/configure-idp.md [hunk 12]
```diff
@@ -0,0 +1,120 @@
...
+
+### Step 4: Assign users to your Okta application
+
+On the page for your Okta app integration, open the **Assignments** tab. Ensure that all users who will use Astronomer are assigned to the integration. For more information, read [Okta Documentation](https://help.okta.com/en/prod/Content/Topics/users-groups-profiles/usgp-assign-apps.htm).
```
Comment (paolaperaza) [inline cloud/configure-idp.md:67]: Should we morph this into a short "Test Integration" step, sort of like Databricks does it here? https://docs.databricks.com/administration-guide/users-groups/single-sign-on/okta.html#test-the-configuration

File: cloud/configure-idp.md [hunk 13]
```diff
@@ -0,0 +1,120 @@
...
+
+## Configure Azure AD as your IDP
```
Comment (paolaperaza) [inline cloud/configure-idp.md:69]:
```suggestion
## Configure Azure AD as your Identity Provider
```

File: cloud/configure-idp.md [hunk 14]
```diff
@@ -0,0 +1,120 @@
...
+
+This section provides setup steps for setting up Azure AD as your IDP on Astronomer Cloud. After completing this setup, your organization's users can use Azure AD to log in to Astronomer.
```
Comment (paolaperaza) [inline cloud/configure-idp.md:71]:
```suggestion
This section provides setup steps for setting up Azure AD as your IdP on Astronomer Cloud. After completing this setup, your organization's users can use Azure AD to log in to Astronomer.
```

File: cloud/configure-idp.md [hunk 15]
```diff
@@ -0,0 +1,120 @@
...
+
+### Prerequisites
+
+To integrate Azure as your IDP for Astronomer you must have:
+
+- An Azure subscription.
+- An [Azure AD tenant](https://docs.microsoft.com/en-us/azure/active-directory/develop/quickstart-create-new-tenant) with `Global Administrator` privileges.
+
+### Step 1: Register your app for Astronomer
```
Comment (paolaperaza) [inline cloud/configure-idp.md:80]:
```suggestion
### Step 1: Register Astronomer as an Application on Azure
```

File: cloud/configure-idp.md [hunk 16]
```diff
@@ -0,0 +1,120 @@
...
+Follow [Microsoft's documentation](https://docs.microsoft.com/en-us/azure/active-directory/develop/quickstart-configure-app-access-web-apis#add-permissions-to-access-web-apis) to add the following **Delegated** permissions to **Microsoft Graph**:
+
+- `OpenId.email`
+- `OpenId.openid`
+- `OpenId.profile`
+- `User.Read`
+
+:::info
+
+If your Azure Active Directory is configured to require admin approval on API permissions, you need additionally click the **Grant admin consent** button at the top of your permissions list.
```
Comment (paolaperaza) [inline cloud/configure-idp.md:108]:
```suggestion
If your Azure Active Directory is configured to require admin approval on API permissions, make sure to also click the **Grant admin consent** button at the top of your permissions list.
```

File: cloud/configure-idp.md [hunk 17]
```diff
@@ -0,0 +1,120 @@
...
+
+### Step 4: Provide Astronomer with your Azure AD app information
+
+Reach out to Astronomer support and provide the following information from your Azure AD app:
+
+- **Microsoft Azure AD Domain**: Retrieve this from your Azure AD directory's overview page in the Microsoft Azure portal.
+- **Application (client) ID**: Retrieve this from the **Overview** page of your application.
+- **Client secret**: Use the value of your client secret from Step 2.
+
+From here, Astronomer will complete the integration and add Azure as your organization's IDP.
```
Comment (paolaperaza) [inline cloud/configure-idp.md:120]:
```suggestion
Reach out to [Astronomer Support](https://support.astronomer.io) and provide the following information from your Azure AD application:

- **Microsoft Azure AD Domain**: Retrieve this from your Azure AD directory's overview page in the Microsoft Azure portal.
- **Application (client) ID**: Retrieve this from the **Overview** page of your application.
- **Client secret**: Use the value of your client secret from Step 2.

From here, Astronomer will complete the integration and add Azure as your organization's IDP.
```

File: cloud/configure-idp.md [hunk 18]
```diff
@@ -0,0 +1,120 @@
...
+- `OpenId.profile`
+- `User.Read`
+
+:::info
+
+If your Azure Active Directory is configured to require admin approval on API permissions, you need additionally click the **Grant admin consent** button at the top of your permissions list.
+
+:::
+
+### Step 4: Provide Astronomer with your Azure AD app information
```
Comment (paolaperaza) [inline cloud/configure-idp.md:112]: Should we add a step 5 to test this? Or is there at least an equivalent to `Step 4: Assign users to your Okta application`?

Wouldn't be the worst thing to have a screenshot of "Okta" or "Azure AD" on the Astronomer login page?

**Review by paolaperaza — COMMENTED** [review]
A few comments here to polish this up

File: cloud/configure-idp.md (hunk 18 above)
Comment (jwitz) [reply cloud/configure-idp.md:112]: @chomipi88 What is the behavior once an IdP is configured: Are users automatically directed to the configured IdP login after specifying their email, or do they go to the login page here and see their IdP listed next to Google/ GH?
Screen Shot 2022-02-04 at 3 54 46 PM

File: cloud/configure-idp.md (hunk 5 above)
Comment (jwitz) [reply cloud/configure-idp.md:16]: I think this is actually related to the last step here, which is to add users to Astronomer via the IdP. As long as they're added to the app, their email will be registered and they can be invited to any Workspace regardless of their base domain.

File: cloud/configure-idp.md (hunk 10 above)
Comment (jwitz) [reply cloud/configure-idp.md:34]: I think this is a setting, not a specific email. It's saying that the default username for the application is the user's email.

**Comment by jwitz** [comment]
@paolaperaza Just made edits and confirmed everything looks good in the preview site. I also think I was able to answer most of your questions directed at @chomipi88 , LMK what ya think!

File: cloud/configure-idp.md [hunk 19]
```diff
@@ -0,0 +1,146 @@
...
+
+This guide provides setup steps for integrating both of these identity providers on Astronomer. Once you complete the integration for your organization:
+
+- Astronomer will be listed in your IdP's set of supported applications.
+- Workspace Admins can invite a user to a Workspace directly without that user having to separately sign up for an account first.
+- Users will automatically be authenticated to Astronomer if they're already logged in to your IdP.
+
+## Configure Okta as Your Identity Provider
+
+This section provides setup steps for setting up Okta as your IDP on Astronomer Cloud. After completing this setup, all users in your organization can use Okta to log in to Astronomer.
```
Comment (paolaperaza) [inline cloud/configure-idp.md:27]:
```suggestion
This section provides setup steps for setting up Okta as your IdP on Astronomer Cloud. After completing this setup, all users in your organization can use Okta to log in to Astronomer.
```

@jwitz Can you make sure we use `IdP` consistently over `IDP` and add to style guide: https://www.notion.so/astronomerio/Astronomer-Terms-and-Technical-Jargon-5d2389d462734e149d2e66a5cf606228

(We'll clean that up later but just so we don't forget)

File: cloud/configure-idp.md (hunk 10 above)
Comment (paolaperaza) [reply cloud/configure-idp.md:50]: Oh word

**Review by paolaperaza — APPROVED** [review]
LGTM @jwitz ! Since Piotr is out, let's move forward with publishing this PR and create a new docs GH issue that we can do later and covers -

- Details on how users now log into Astronomer with Okta or Azure AD combined (content & screenshots or gif)
- Steps to test either integration

File: cloud/configure-idp.md (hunk 18 above)
Comment (chomipi88) [reply cloud/configure-idp.md:132]: @jwitz If their organization has multiple connection types, they would see another option under Continue with GitHub. If their organization is only configured with Okta or Azure AD, the user would never see that screen and be redirected to their IdP

File: cloud/configure-idp.md (hunk 5 above)
Comment (chomipi88) [reply cloud/configure-idp.md]: @paolaperaza Yes the email domain configured for the organization should match the email domain the users are logging in with. Today only one domain is supported, but we are working on multi domain support.

**Pull request merged**